kernel/poll.c

```diff
@@ -197,6 +231,10 @@ int k_poll(struct k_poll_event *events, int num_events, s32_t timeout)
 	int last_registered = -1, rc;
 	unsigned int key;
 
+	if (num_events == 1) {
+		return poll_1(events, timeout);
```

Comment:
why not just a goto to a label on line 241? I am not sure introducing poll_1() optimize that much the path.